controllers/topology/internal/contract/controlplane.go

```diff
@@ -93,6 +138,31 @@ func (v *ControlPlaneVersion) Set(obj *unstructured.Unstructured, value string)
 	return nil
 }
 
+// ControlPlaneStatusVersion provide a helper struct for working with version in ClusterClass.
+type ControlPlaneStatusVersion struct{}
+
+// Path returns the path of the reference.
+func (v *ControlPlaneStatusVersion) Path() Path {
+	return Path{"status", "version"}
+}
+
+// Get gets the version value from the ControlPlane object.
+func (v *ControlPlaneStatusVersion) Get(obj *unstructured.Unstructured) (*string, error) {
+	value, ok, err := unstructured.NestedString(obj.UnstructuredContent(), v.Path()...)
+	if err != nil {
+		return nil, err
+	}
+	if !ok {
+		return nil, errors.Wrapf(errNotFound, "path %s", "."+strings.Join(v.Path(), "."))
```

Comment:
I checked and this approach will not work when we want to do checks like `errors.Is(err, errNotFound)` - we want to be able to do these kinds of checks to check if the field that we are looking for are set in the status of certain objects.  
 
The error message will have all the details but will not read like a proper english sentence.

Instead of :
* `some wrapping error message: path .a.b.c not found`

we get :
* `some wrapping message: path .a.b.c: not found`


IMO, i think the way the error message reads is fine.
WDYT?

Reply:
ok to keep as it is, we can iterate later eventually.
Let's make use to use errNotFound consistently in all the Get methods (also the existing ones)